connectors/byoc.py

```diff
@@ -309,14 +305,12 @@ def __init__(self, filter_=None):
 
         super().__init__(filter_)
 
-        advanced_rules = filter_.get("advanced_snippet", {})
-
-        self.advanced_rules = advanced_rules.get("value", advanced_rules)
+        self.advanced_rules = filter_.get("advanced_snippet", {})
         self.basic_rules = filter_.get("rules", [])
         self.validation = filter_.get("validation", {"state": "", "errors": []})
 
     def get_advanced_rules(self):
-        return self.advanced_rules
+        return self.advanced_rules.get("value", self.advanced_rules)
```

Comment:
So we return content of "advanced_snippet.value" if there's anything inside, otherwise content of "advanced_snippet"?

Reply:
It's probably better to return `{}` as default. Shouldn't happen anyway, but it's more obvious, that "value" doesn't exist.

Reply:
Yeah I saw original defaulting logic used `{}`, probably indeed better default to it

Reply:
Adjusted it 👍